tree.go

```diff
@@ -388,7 +388,9 @@ func NewRoster(ids []*network.ServerIdentity) *Roster {
 		agg := ids[0].Public.Clone()
 		agg.Null()
 		for _, e := range ids {
-			agg = agg.Add(agg, e.Public)
+			if e.Public != nil {
+				agg = agg.Add(agg, e.Public)
+			}
```

Comment:
can `e.Public` be nil?  My guess is that it can't because `ids` should have actual content, right? Maybe we need to return a nil roster in this case.